### DISCUSSION TITLE
Proper Linux environment, not using the existing Linux terminal.

### DISCUSSION THREAD

**Discussion by techguy16** [PowerShell/PowerShell#19835] [Enhancements]
I think it would be nice if there was an official 'PowerShell' command line, customized for PowerShell on Linux. It's just an idea.

**Comment by jdhitsolutions** [comment]
Do you mean command prompt? What do you have in mind that would make it 'Linux'?

**Reply by techguy16** [reply]
What do you mean? Are you meaning Linux-specific stuff?

**Reply by rhubarb-geek-nz** [reply]
What are you expecting to be different?

pwsh already executes as a command line application, just like bash, dash, sh, zsh, csh, ksh etc.

**Reply by techguy16** [reply]
Make it like Windows Terminal, with the ability to detect installed CLIs and a simple way to switch between them, and of course, customizable themes.

**Reply by rhubarb-geek-nz** [reply]
Why? You are running on Linux so things should be done the Linux way. If you want things done like Windows then run on Windows.
If you want a fancy UI or IDE for PowerShell on Linux then use Visual Studio Code.

**Reply by techguy16** [reply]
I'm not talking about for me, I'm talking about it for new Linux users.

**Reply by rhubarb-geek-nz** [reply]
I recommend that new Linux users should understand how the components of Linux work together, where each component does one thing well. In the case of PowerShell it's job is to run commands and scripts. The features you are describing belong as part of a GUI program, like gnome-terminal, konsole, dtterm, xterm etc.

**Reply by techguy16** [reply]
Well, if everyone is so against this idea, I'll just ignore it. I was going to develop this myself, but I'm not going to touch it at all.

**Reply by rhubarb-geek-nz** [reply]
You are welcome to build it if you think there is a need. My question was really about what is it that you think you are wanting to build. From what you describe it is not PowerShell specific at all, it sounds like a general GUI terminal that is able to customise itself based on multiple different shells that are installed on the system. Then the next step is what would be the starting point for such a project given it would be competing with the terminal programs I already listed, eg kconsole and gnome-terminal and would need to integrate well on a KDE, Gnome or any other desktop.

**Reply by rhubarb-geek-nz** [reply]
Depending on the use cases for the new user, for example if they are familiar with Windows but not Linux, or you are expecting them to write PowerShell scripts then I suggest that Visual Studio Code is already that tool.

- already exists on Windows, Linux and MacOS with consistent interface between them
- has extensions to support PowerShell and many other technologies
- supports syntax highlighting for PowerShell scripts
- supports editing multiple types of text files

**Reply by techguy16** [reply]
It's a code editor, not a thing like Windows Terminal.

**Reply by ThomasNieto** [reply]
Since it seems like you're wanting something like Windows Terminal for Linux. Have you tried [Tabby](https://tabby.sh)? I have not used it personally but looks like it has similar features to Windows Terminal.

Here is an excerpt from the [GitHub](https://github.com/eugeny/tabby) project.

> [Tabby](https://tabby.sh/) (formerly Terminus) is a highly configurable terminal emulator, SSH and serial client for Windows, macOS and Linux

- Integrated SSH and Telnet client and connection manager
- Integrated serial terminal
- Theming and color schemes
- Fully configurable shortcuts and multi-chord shortcuts
- Split panes
- Remembers your tabs
- PowerShell (and PS Core), WSL, Git-Bash, Cygwin, MSYS2, Cmder and CMD support
- Direct file transfer from/to SSH sessions via Zmodem
- Full Unicode support including double-width characters
- Doesn't choke on fast-flowing outputs
- Proper shell experience on Windows including tab completion (via Clink)
- Integrated encrypted container for SSH secrets and configuration
- SSH, SFTP and Telnet client available as a [web app](https://tabby.sh/app) (also [self-hosted](https://github.com/Eugeny/tabby-web)).

**Comment by kasini3000** [comment]
1 Now, you can set Powershell as your default shell for Linux.
2 Setting the powershell as the default shell on the production server is currently a challenge.
3 Because 1 and 2 , Please learn about my powershell script framework “Kasini3000” in my repo.
Powershell is the best engine in the world, and I used it to make the best cars. 
With Kasini3000, you don't need to use the powershell as the default shell to use all the benefits of powershell.

### Specific:
* Not change linux default shell, powershell on remote linux exec by sshd.
* It can add the ability to remotely pass objects without changing the default Linux shell.
```powershell
$variable_from_node_to_master = krun { get-date }
$variable_from_node_to_master.year
#----------
$object_from_master_to_node = get-date
$a = 1 
krun { $args[0].year + $args[1] } -a $object_from_master_to_node,$a
# return 2024
```

* It allows you to remotely pass awk, sed code blocks without considering single quotes or double quotes escaping.

```powershell
$v2 = k_run_ip -ip 1.2.3.4 {

	$bashcmd = @'
	netstat -an | awk ' /LISTEN/{next};/^tcp/{s=split($5, N,":"); \
	A[N[s-1]]++} END{ for (a in A) print a, A[a]} '
	'@

	${bashcmd} | /bin/bash

}
```

# Powershell gives your “Linux node” a pair of wings!

**Reply by kasini3000** [reply]
note:title changed but no change history.
This incident has taught me a lesson. In my future posts, I should keep and quote the original post.

According to my understanding of the initial title, poster pursues the feasibility of using powershell as the default shell.
